swarm/netsim/events_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package netsim
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestPeerEvents(t *testing.T) {
```

Comment:
Please add a short description for what the test does if the test does a number of steps, e.g.
- Create simulation
- Add 2 nodes
- Register for PeerEvents
- Connect peers in chain
- Check amount of expected events  received, should match

Not needed for all tests of course (there are tests which really it is easy to understand, like `ExampleSimulation_WaitTillHealthy`